src/abi_impl.rs

```diff
@@ -0,0 +1,69 @@
+use crate::env::{get_remaining_points_for_env, sub_remaining_point, Env};
+use crate::settings;
+use crate::types::{Address, Response};
+use anyhow::Result;
+use as_ffi_bindings::{Read as ASRead, StringPtr, Write as ASWrite};
+
+/// `Call` ABI called by the webassembly VM
+///
+/// Call an exported function in a WASM module at a given address
+///
+/// It take in argument the environment defined in env.rs
+/// this environment is automatically filled by the wasmer library
+/// And two pointers of string. (look at the readme in the wasm folder)
+fn call_module(env: &Env, address: &Address, function: &str, param: &str) -> Result<Response> {
+    sub_remaining_point(env, settings::metering_call()).unwrap();
+    let module = &env.interface.get_module(address).unwrap();
+    crate::execution_impl::exec(
+        get_remaining_points_for_env(env),
+        None,
+        module,
+        function,
+        param,
+        &*env.interface,
+    )
+}
+
+/// Raw call that have the right type signature to be able to be call a module
+/// directly form AssemblyScript:
+///
```

Comment:
TODO : remove all the unwrap() when we are exposed to user code. The SC can send arbitrary data here and we shouldn't crash

Reply:
maybe let's open a followup issue on removing all the unsafe unwrap/expect

Reply:
@damip, you can note that the unwrap in function called only by wasmer (with the ImportObject) has to be called to signal to the wasmer instance that something got wrong and doesn't make the full software panic.

Unless you use `run(...).unwrap() it shouldn't make the app panic. I can add a kind of unit test to be sure of that!!

Reply:
look at this issue! :wink: https://github.com/massalabs/sc-runtime/issues/45 we can continue forward the error management in another PR.

Reply:
Perfetto